benchmarks/experiment_runner.py

```diff
@@ -276,8 +276,7 @@ def collect_profile_to_metrics(self, prof, metrics):
       if evt.device_type == DeviceType.CPU:
         # in legacy profiler, kernel info is stored in cpu events
         if evt.is_legacy:
-          if not use_device:
```

Comment:
`use_device` wasn't defined anywhere.

Reply:
Thanks for noticing this. It means that this path was never exercised before then. Can you raise an exception here instead so we won't be overcounting the profiler time if something changes?

Reply:
I did end up hitting this branch. As far as I remember, accounting for this number is correct. Let me take a look at how PyTorch counts that to confirm.

Reply:
On what model did you hit this branch?

Reply:
It might be GPU specific then.  Can you double check that there's not overcounting, post it here, and then we will merge to the main branch?

Reply:
This is similar to how PyTorch summarizes its data ([link](https://github.com/pytorch/pytorch/blob/afe6d272c69ae5671ca0df978be8fff7e8e4ed4e/torch/autograd/profiler_util.py#L1025-L1032)). Although it does explicitly checks for `use_device`, it's a redundant check, since [it's already done when we access the attribute](https://github.com/pytorch/pytorch/blob/afe6d272c69ae5671ca0df978be8fff7e8e4ed4e/torch/autograd/profiler_util.py#L567-L576). i.e. if `use_device == True`, `evt.self_cuda_time_total` will be 0.

Apparently, we execute the `not use_device` branch when PyTorch is compiled without support for Kineto.
That said, I ran `alexnet` and didn't notice any signs of overcounting.

Reply:
Thanks for checking this!